projects/core/src/cms/services/cms-component.loader.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { CmsComponent } from '../../occ/occ-models/index';
+import { PageContext } from '../../routing/models/page-context.model';
+import { CmsComponentAdapter } from './cms-component.adapter';
+import { CmsStructureConfigService } from './cms-structure-config.service';
+
+/**
+ * Abstract class that can be used to implement custom loader logic
+ * in order to load CMS components from third-party CMS system.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export abstract class CmsComponentLoader<T> {
+  constructor(
+    protected cmsStructureConfigService: CmsStructureConfigService,
+    protected adapter: CmsComponentAdapter<T>
+  ) {}
+
+  /**
+   * Abstract method must be used to load the component for a given `id` and `PageContext`.
+   * The component can be loaded from alternative backend, as long as the structure
+   * converts to the `CmsStructureModel`.
+   *
+   * @param pageContext The `PageContext` holding the page Id.
+   */
+  abstract load(
+    _id: string,
```

Comment:
Is there a reason for underscores for those parameters?

Reply:
I had that to suppress the tslint warnings, but that was before I made it abstract, thx!